### PR TITLE
Fix various spelling errors

### PR DESCRIPTION
**Pull request by shawntabrizi**
Closes: https://github.com/paritytech/substrate/issues/4933

I used https://github.com/streetsidesoftware/cspell

with the following configuration:

```js
// cSpell Settings
{
	// Version of the setting file.  Always 0.1
	"version": "0.1",
	// language - current active spelling language
	"language": "en",
	// words - list of words to be always considered correct
	"words": [
		"serde",
		"dispatchable",
		"perbill",
		"hasher",
		"unbonding",
		"unbond",
		"rebond",
		"unbonded",
		"stakers",
		"validator",
		"validator's",
		"staker",
		"phragmen",
		"uvec",
		"prefs",
		"unstake",
		"misbehaviors",
		"slashable",
		"uint",
		"invulnerables",
		"npos",
		"mutables",
		"unlookup",
		"twox",
		"gavofyork",
		"consts",
		"overslash",
		"incentivizes",
		"structfield",
		"powf",
		"inpos",
		"unstaked",
		"testnets",
		"doesnt",
		"sesh",
		"referencer",
		"overslashing",
		"subtrait",
		"bitor",
		"unreserving",
		"staking's",
		"suspendable",
		"preimage",
		"undelegate",
		"cooloff",
		"proxying",
		"preimages",
		"undelegated",
		"thresholding",
		"undelegation",
		"supermajority",
		"enumflags",
		"concat",
		"passthrough",
		"wasm",
		"disincentivize",
		"permissionless",
		"decodable",
		"permill",
		"dispatchable",
		"deserializer",
		"serializer",
		"leaderboard",
		"whatevs",
		"wabt",
		"pwasm",
		"entrypoints",
		"funcref",
		"deserialized",
		"reserialized",
		"BACKTRACE",
		"entrypoint",
		"unhashed",
		"serializable",
		"newtype",
		"offchain",
		"imonline",
		"pubkey",
		"heartbeat",
		"inherents",
		"runtimes",
		"unvested",
		"utilisuba",
		"modlpy",
		"sigs",
		"timepoint",
		"multisigs",
		"keypair",
		"stateful",
		"suba",
		"multisig",
		"Pseudonymal",
		"prng",
		"wasmi",
		"camelcase",
		"extrinsics",
		"nocompile",
		"runtime's",
		"handleable",
		"totokens",
		"idents",
		"parens",
		"bitmask",
		"hashable",
		"appendable",
		"uncheckedextrinsic",
		"structs",
		"debuggable",
		"scrate",
		"impls",
		"LINKEDMAPU",
		"PUBGETLINKEDMAPU",
		"COMPLEXTYPE",
		"unstaking",
		"trybuild",
		"WITHCONFIG",
		"PUBU",
		"testextension",
		"dispatchables",
		"enum's",
		"partialeq",
		"Weighable",
		"hiddencrate",
		"whitespaces",
		"hiddencrate",
		"claused",
		"getfn",
		"sline",
		"tokenstream",
		"WITHCONFIGMYDEF",
		"PUBMAPU",
		"PUBGETMAPU",
		"MAPU",
		"GETMAPU",
		"GETLINKEDMAPU",
		"PUBLINKEDMAPU",
		"pubmap",
		"mydef",
		"extragenesis",
		"paren",
		"coldot",
		"untrusted",
		"dser",
		"hashers",
		"prepended",
		"traitinstance",
		"GETU",
		"PUBGETU",
		"WITHCONFIGMYDEFOPT",
		"btree",
		"subtrie",
		"unmarshall",
		"enaction",
		"preload",
		"Computor",
		"rustc",
		"recurse",
		"bytecode",
		"parameterizations",
		"accountid",
		"instantiator",
		"toks",
		"sandboxing",
		"codehash",
		"reinstrument",
		"memorywise",
		"sandboxed",
		"unmarshalling",
		"reinstrumentation",
		"anyfunc",
		"forgetit",
		"subaccounts",
		"judgements",
		"Keccak",
		"memoized",
		"blocknumber",
		"revoting",
		"inspectable",
		"sudid",
		"sudoer",
		"unvouch",
		"unfound",
		"unbidding",
		"rewardees",
		"underflowed",
		"Applyable",
		"attackable",
		"bvalue",
		"precompile",
		"precompiles",
		"ethereum",
		"gameability",
		"panickers",
		"myname",
		"Dapps",
		"merkle",
		"blockchain",
		"DEDUP",
		"Unrequested",
		"subaccount",
		"headshot",
		"doctests",
		"backticks",
		"snapshotted",
		"reproposing",
		"rposition",
		"Rustdocs",
		"rustdoc",
		"data's",
		"struct's",
		"endianness",
		"encodable",
		"retv",
		"coinbase",
		"nonfull",
		"Unscored",
		"vals",
		"chacha",
		"incentivized",
		"Malloc",
		"keystore",
		"seedable",
		"blockchains",
		"unfounding",
		"unvouched",
		"extrinsic's",
		"tx's",
		"turbofish",
		"biguint",
		"lpad",
		"webassembly",
		"there'll",
		"polkadot",
		"memtrie",
		"unpopulate",
		"memdb",
		"keyspace",
		"txpool",
		"genericity",
		"standardmap",
		"uniqueid",
		"Schnorrkel",
		"preaudit",
		"descendents",
		"unappliable",
		"perquintill",
		"lstrip",
		"rhat",
		"qhat",
		"honggfuzz",
		"biguints",
		"unergonomic",
		"HFUZZ",
		"vecs",
		"suri",
		"Ristretto",
		"Schnorr",
		"Multiaddr",
		"hexdisplay",
		"deser",
		"rustversion",
		"footgun",
		"Inflector",
		"Uninit",
		"ecdsa",
		"destructured",
		"parachain",
		"superset",
		"Ferdie",
		"dealloc",
		"nanos",
		"secp",
		"alloc",
		"libsecp",
		"finalnum",
		"deserialization",
		"byteorder",
		"dalek",
		"HDKD",
		"zeroize",
		"hmac",
		"microcontrollers",
		"rngs",
		"Cosignature",
		"fuzzer",
		"musig",
		"bigint",
		"wasi",
		"propagations",
		"UUID",
		"backends",
		"testchild",
		"childtrie",
		"childchange",
		"schnorrkel's",
		"impl'd",
		"impl'ing",
		"castable",
		"exts",
		"RPCs",
		"repr",
		"strat",
		"ptrs",
		"supertraits",
		"deallocation",
		"dogglesworth",
		"std's",
		"deque",
		"keyspaced",
		"semver",
		"bytesref",
		"drilldown",
		"externval",
		"funcs",
		"deserializable",
		"Unserializable",
		"Serializers",
		"testval",
		"timespan",
		"heappages",
		"memref",
		"endians",
		"bytewise",
		"testinh",
		"downcasting",
		"mainnet",
		"libc",
		"Unallowed",
		"subkey",
		"mkdir",
		"wasmtime",
		"bindgen",
		"rocksdb",
		"structopt",
		"tempfile",
		"multiaddress",
		"MILLISECS",
		"bootnodes",
		"testnet",
		"unlicense",
		"bootnode",
		"subcommand",
		"chainspec",
		"subcommand",
		"toolchain",
		"rustup",
		"vergen",
		"auths",
		"hardcoded",
		"deduplication",
		"dref",
		"asciish",
		"autocfg",
		"memmap",
		"getrandom",
		"bitvec",
		"arrayvec",
		"winapi",
		"pnames",
		"ptypes",
		"pborrow",
		"threadsafe",
		"auraslot",
		"babeslot",
		"dword",
		"plugable",
		"struct",
		"destinated",
		"substream",
		"mitigations",
		"RUSTFLAGS",
		"OPENSSL",
		"USERPROFILE",
		"LLVM",
		"vcpkg",
		"libssl",
		"libclang",
		"cmake",
		"cryptocurrency",
		"roadmap",
		"Pessimization",
		"mychain",
		"codeowner",
		"breaksapi",
		"pleasereview",
		"blockchain's",
		"symlink",
		"buildtools",
		"adoc",
		"breaksconsensus",
		"openish",
		"customizability",
		"PBFT",
		"DDOS",
		"noncanonical",
		"canonicalization",
		"canonicality",
		"uncanonicalized",
		"uncanonicalize",
		"canonicalized",
		"canonicalize",
		"canonicalizes",
		"memorydb",
		"kvdb",
		"forkable",
		"unqueue",
		"oneshot",
		"rebalance",
		"unimported",
		"finalizer",
		"testsuite",
		"BACKOFF",
		"rebalance",
		"rebalancing",
		"precommit",
		"prevote",
		"precommits",
		"prevoted",
		"unfinalized",
		"resetted",
		"invalidators",
		"heapsize",
		"onchain",
		"prevotes",
		"precommitted",
		"cors",
		"bitflags",
		"rtype",
		"Grafana",
		"SIGTERM",
		"SIGTERM",
		"SIGINT",
		"peerset",
		"quickcheck",
		"peerset",
		"proto",
		"protos",
		"strs",
		"cmds",
		"mdns",
		"protobuf",
		"preruntime",
		"checked",
		"compat",
		"yamux",
		"subcommands",
		"tempdir",
		"Kademlia",
		"flattenable",
		"peersstate",
		"threadpool",
		"nfin",
		"unpruned",
		"Unroutable",
		"smallvec",
		"curr",
		"errno",
		"ctrie",
		"rpassword",
		"isatty",
		"inout",
		"CHTs",
		"ucht",
		"lmap",
		"kinded",
		"fdlimit",
		"journaled",
		"deregistered",
		"unregisters",
		"uninserted",
		"interruptible",
		"errored",
		"reenable",
		"syscalls",
		"syscall",
		"klen",
		"inmem",
		"dbtx",
		"UNDECODABLE",
		"canonicalizations",
		"prunings",
		"libp",
		"reput",
		"repropagated",
		"repropagate",
		"Muxer",
		"pset",
		"nohash",
		"addrs",
		"numer",
		"kbytes",
		"prost",
		"Kademlia's",
		"Tokio's",
		"uncategorized",
		"mysecret",
		"Notif",
		"mplex",
		"boneh",
		"unapply",
		"comms",
		"repropagation",
		"disp",
		"fprb",
		"cpus",
		"trie's",
		"orgs",
		"abcdefg",
		"txqueue",
		"txcount",
		"fchr",
		"unwatch",
		"unsecure",
		"sysinfo",
		"rustls",
		"sync'ed",
		"connect'em",
		"oneof",
		"foobarbaz",
		"secio",
		"repu",
		"upgr",
		"deps",
		"kbuckets",
		"Bloaty",
		"blocknum",
		"QWERTYUIOPASDFGHJKLZXCVBNM",
		"MILLICENTS",
		"nocapture",
		"feemul",
		""
	],
	"ignoreWords": [
		"initialise",
		"finalise",
		"srml",
		"overlayed",
		"malus",
		"Kianenigma",
		"Amar",
		"joepetrowski",
		"bkchr",
		"Dothereum",
		"kulupu",
		"kusama",
		"edgeware",
		"substratee",
		"paritytech",
		"Emberic",
		"Danta",
		"pepyakin",
		"tomusdrw",
		"andresilva",
		"demi",
		"mxinden",
		"thiolliere",
		"sorpaas",
		"Robbepop",
		"Behaviour",
		"Behaviour's",
		"misbehaviour",
		"Millo",
		"Finalised",
		"UTXO",
		""
	],
	// flagWords - list of words to be always considered incorrect
	// This is useful for offensive words and common spelling errors.
	// For example "hte" should be "the"
	"flagWords": [
		"hte"
	],
	"allowCompoundWords": true
}
```

One word that we should still possibly fix is usage of `Behaviour`, but is shows up in a lot of places and some libraries.

### CODE REVIEW

File: client/finality-grandpa/src/tests.rs [hunk 1]
```diff
@@ -982,7 +982,7 @@ fn force_change_to_new_set() {
 
 	// it will only finalize if the forced transition happens.
 	// we add_blocks after the voters are spawned because otherwise
-	// the link-halfs have the wrong AuthoritySet
+	// the link-halveshave the wrong AuthoritySet
```
Comment (shawntabrizi) [inline client/finality-grandpa/src/tests.rs]:
```suggestion
	// the link-halves have the wrong AuthoritySet
```

**Comment by kianenigma** [comment]
> Might consider also to add this to CI.

+1

File: bin/node/cli/tests/common.rs [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 use std::{process::{Child, ExitStatus}, thread, time::Duration};
 
-/// Wait for the given `child` the given ammount of `secs`.
+/// Wait for the given `child` the given amount of `secs`.
```
Comment (joepetrowski) [inline bin/node/cli/tests/common.rs]:
```suggestion
/// Wait for the given `child` the given number of `secs`.
```

File: client/api/src/execution_extensions.rs [hunk 3]
```diff
@@ -64,7 +64,7 @@ impl Default for ExecutionStrategies {
 
 /// Generate the starting set of ExternalitiesExtensions based upon the given capabilities
 pub trait ExtensionsFactory: Send + Sync {
-	/// Make `Extensions` for given Capapbilities
+	/// Make `Extensions` for given Capabilities
```
Comment (joepetrowski) [inline client/api/src/execution_extensions.rs]:
```suggestion
	/// Make `Extensions` for given `Capabilities`.
```

File: client/api/src/execution_extensions.rs [hunk 4]
```diff
@@ -77,7 +77,7 @@ impl ExtensionsFactory for () {
 /// A producer of execution extensions for offchain calls.
 ///
 /// This crate aggregates extensions available for the offchain calls
-/// and is responsbile to produce a right `Extensions` object
+/// and is responsible to produce a right `Extensions` object
```
Comment (joepetrowski) [inline client/api/src/execution_extensions.rs]: Dumb question, do you mean "correct" object? I would use that unless this is referring to a right/left distinction.
```suggestion
/// and is responsible for producing a right `Extensions` object.
```

File: client/cli/src/params.rs [hunk 5]
```diff
@@ -718,7 +718,7 @@ pub struct BuildSpecCmd {
 	pub node_key_params: NodeKeyParams,
 }
 
-/// Wrapper type of `String` which holds an arbitary sized unsigned integer formatted as decimal.
+/// Wrapper type of `String` which holds an arbitrary sized unsigned integer formatted as decimal.
```
Comment (joepetrowski) [inline client/cli/src/params.rs]:
```suggestion
/// Wrapper type of `String` that holds an unsigned integer of arbitrary size, formatted as a decimal.
```

File: client/executor/common/src/sandbox.rs [hunk 6]
```diff
@@ -498,7 +498,7 @@ pub fn instantiate<'a, FE: SandboxCapabilities>(
 ///
 /// This is generic over a supervisor function reference type.
 pub struct Store<FR> {
-	// Memories and instances are `Some` untill torndown.
+	// Memories and instances are `Some` until torndown.
```
Comment (joepetrowski) [inline client/executor/common/src/sandbox.rs]:
```suggestion
	// Memories and instances are `Some` until torn down.
```

File: client/finality-grandpa/src/communication/mod.rs [hunk 7]
```diff
@@ -153,14 +153,14 @@ pub(crate) struct NetworkBridge<B: BlockT, N: Network<B>> {
 
 	/// `NeighborPacketWorker` processing packets sent through the `NeighborPacketSender`.
 	//
-	// `NetworkBridge` is required to be clonable, thus one needs to be able to clone its children,
-	// thus one has to wrap neighor_packet_worker with an `Arc` `Mutex`.
+	// `NetworkBridge` is required to be cloneable, thus one needs to be able to clone its children,
+	// thus one has to wrap neighbor_packet_worker with an `Arc` `Mutex`.
```
Comment (joepetrowski) [inline client/finality-grandpa/src/communication/mod.rs]:
```suggestion
	// thus one has to wrap `neighbor_packet_worker` with an `Arc` `Mutex`.
```

File: client/state-db/src/pruning.rs [hunk 8]
```diff
@@ -36,7 +36,7 @@ pub struct RefWindow<BlockHash: Hash, Key: Hash> {
 	death_rows: VecDeque<DeathRow<BlockHash, Key>>,
 	/// An index that maps each key from `death_rows` to block number.
 	death_index: HashMap<Key, u64>,
-	/// Block number that corresponts to the front of `death_rows`
+	/// Block number that corresponds to the front of `death_rows`
```
Comment (joepetrowski) [inline client/state-db/src/pruning.rs]:
```suggestion
	/// Block number that corresponds to the front of `death_rows`.
```

File: frame/contracts/src/tests.rs [hunk 9]
```diff
@@ -1052,7 +1052,7 @@ fn claim_surcharge_malus() {
 }
 
 /// Claim surcharge with the given trigger_call at the given blocks.
-/// if removes is true then assert that the contract is a tombstonedead
+/// if removes is true then assert that the contract is a tombstone
```
Comment (joepetrowski) [inline frame/contracts/src/tests.rs]:
```suggestion
/// If `removes` is true then assert that the contract is a tombstone.
```

File: frame/support/src/weights.rs [hunk 10]
```diff
@@ -67,7 +67,7 @@ pub trait ClassifyDispatch<T> {
 	fn classify_dispatch(&self, target: T) -> DispatchClass;
 }
 
-/// Means of determining the weight of a block's lifecycle hooks: on_initialize, on_finalize and
+/// Means of determining the weight of a block's life cycle hooks: on_initialize, on_finalize and
```
Comment (joepetrowski) [inline frame/support/src/weights.rs]:
```suggestion
/// Means of determining the weight of a block's life cycle hooks: `on_initialize`, `on_finalize` and
```

File: primitives/blockchain/src/lib.rs [hunk 11]
```diff
@@ -14,7 +14,7 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Substrate blockchain traits and primtives
+//! Substrate blockchain traits and primitives
```
Comment (joepetrowski) [inline primitives/blockchain/src/lib.rs]:
```suggestion
//! Substrate blockchain traits and primitives.
```

File: primitives/core/src/ed25519.rs [hunk 12]
```diff
@@ -433,7 +433,7 @@ impl TraitPair for Pair {
 
 	/// Make a new key pair from secret seed material.
 	///
-	/// You should never need to use this; generate(), generate_with_phrasee
+	/// You should never need to use this; generate(), generate_with_phrase
```
Comment (joepetrowski) [inline primitives/core/src/ed25519.rs]: This comment could be more clear in general (although maybe not directly related to this PR). Is this saying I should use `generate_with_phrase` instead?

File: primitives/core/src/offchain/mod.rs [hunk 13]
```diff
@@ -335,7 +335,7 @@ pub trait Externalities: Send {
 
 	/// Returns a random seed.
 	///
-	/// This is a trully random non deterministic seed generated by host environment.
+	/// This is a truly random non deterministic seed generated by host environment.
```
Comment (joepetrowski) [inline primitives/core/src/offchain/mod.rs]:
```suggestion
	/// This is a truly random, non-deterministic seed generated by host environment.
```

File: primitives/io/src/lib.rs [hunk 14]
```diff
@@ -594,7 +594,7 @@ pub trait Offchain {
 
 	/// Returns a random seed.
 	///
-	/// This is a trully random non deterministic seed generated by host environment.
+	/// This is a truly random non deterministic seed generated by host environment.
```
Comment (joepetrowski) [inline primitives/io/src/lib.rs]:
```suggestion
	/// This is a truly random, non-deterministic seed generated by host environment.
```

File: client/api/src/execution_extensions.rs (hunk 4 above)
Comment (shawntabrizi) [reply client/api/src/execution_extensions.rs]: Yes I think it means "correct" here.

Comment (shawntabrizi) [inline client/api/src/execution_extensions.rs]:
```suggestion
/// and is responsible to produce a correct `Extensions` object
```

File: client/api/src/execution_extensions.rs [hunk 15]
```diff
@@ -77,7 +77,7 @@ impl ExtensionsFactory for () {
 /// A producer of execution extensions for offchain calls.
 ///
 /// This crate aggregates extensions available for the offchain calls
-/// and is responsbile to produce a right `Extensions` object
+/// and is responsible for producing a right `Extensions` object.
```
Comment (shawntabrizi) [inline client/api/src/execution_extensions.rs]:
```suggestion
/// and is responsible for producing a correct `Extensions` object.
```

**Comment by pepyakin** [comment]
The CI failure is related to #4944

**Pull request merged**